background_scripts/main.coffee

```diff
@@ -655,6 +655,21 @@ sendRequestHandlers =
   gotoMark: Marks.goto.bind(Marks)
   setBadge: setBadge
 
+# We always remove chrome.storage.local/findModeRawQueryListIncognito on startup.
+chrome.storage.local.remove "findModeRawQueryListIncognito"
+
+# Remove chrome.storage.local/findModeRawQueryListIncognito if there are no remaining incognito-mode tabs.
+# Since the common case is that there are none to begin with, we first check whether the key is set at all.
+chrome.tabs.onRemoved.addListener (tabId) ->
+  chrome.storage.local.get "findModeRawQueryListIncognito", (items) ->
+    if items.findModeRawQueryListIncognito
+      chrome.windows.getAll { populate: true }, (windows) ->
+        for window in windows
+          for tab in window.tabs
```

Comment:
We can query `window.incognito` directly, so we don't have to loop over all the tabs (or use `populate: true` above).